pkg/engine/template_generator.go

```diff
@@ -1022,3 +1124,76 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"shellQuote": helpers.ShellQuote,
 	}
 }
+
+func (t *TemplateGenerator) GenerateTemplateV2(containerService *api.ContainerService, generatorCode string, acsengineVersion string) (templateRaw string, parametersRaw string, err error) {
+
+	armParams, _ := t.getParameterDescMap(containerService)
+	armResources := GenerateARMResources(containerService)
+	armVariables := GetKubernetesVariables(containerService)
+	armOutputs := GetKubernetesOutputs(containerService)
+
+	armTemplate := ARMTemplate{
+		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+		ContentVersion: "1.0.0.0",
+		Parameters:     armParams["parameters"],
+		Variables:      armVariables,
+		Resources:      armResources,
+		Outputs:        armOutputs,
+	}
+
+	templBytes, _ := json.Marshal(armTemplate)
+	templateRaw = string(templBytes)
+
+	var parametersMap paramsMap
+	if parametersMap, err = getParameters(containerService, generatorCode, acsengineVersion); err != nil {
+		return "", "", err
+	}
+
+	var parameterBytes []byte
+	if parameterBytes, err = helpers.JSONMarshal(parametersMap, false); err != nil {
+		return "", "", err
+	}
+	parametersRaw = string(parameterBytes)
+
+	return templateRaw, parametersRaw, err
+}
+
+func (t *TemplateGenerator) getParameterDescMap(containerService *api.ContainerService) (map[string]interface{}, error) {
+	var templ *template.Template
+	var paramsDescMap map[string]interface{}
+	properties := containerService.Properties
+	// save the current orchestrator version and restore it after deploying.
+	// this allows us to deploy agents on the most recent patch without updating the orchestator version in the object
```

Comment:
typo on orchestrator here

Reply:
Pre-existing. Will fix that upstream.

Reply:
oh actually nvm. I moved it there. I will fix it in this PR :)